Add dn based output trigger

pgrete opened the pull request:
<!--Provide a general summary of your changes in the title above, for
example "Add AMR unit test for cell centered fields.".  Please avoid
non-descriptive titles such as "Addresses issue #8576".-->

## PR Summary

What is says on the can (plus removed dead vtk code as I don't think we'll ever implement/reactivate it).

Still need
- [x] doc
- [x] changelog
- [ ] tests?

<!--If it fixes an open issue, please link to the issue here.-->

## PR Checklist

<!-- Note that some of these check boxes may not apply to all pull requests -->

- [ ] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Code is formatted
- [ ] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] Docs build
- [ ] (@lanl.gov employees) Update copyright on changed files

pgrete committed:
Add dn based output trigger

BenWibking commented:
I can't tell whether the macOS CI failure is related or not:
```
The following tests FAILED:
	 68 - regression_test:boundary_exchange (Failed)
	 78 - regression_test:poisson (Failed)
```
It looks like some of the output files don't exist, so it might be.

pgrete commented:
> I can't tell whether the macOS CI failure is related or not:
> 
> ```
> The following tests FAILED:
> 	 68 - regression_test:boundary_exchange (Failed)
> 	 78 - regression_test:poisson (Failed)
> ```
> 
> It looks like some of the output files don't exist, so it might be.

I think the updated handling of ensuring a last time based output is written might be broken (or not behaving as before any more).
I'm investigating.

Yurlungur reviewed (approved):
Assuming we can make tests pass, LGTM. :+1: for deleting dead code.

BenWibking committed:
Merge branch 'develop' into pgrete/dncycle

pgrete committed:
Merge branch 'develop' into pgrete/dncycle

pgrete committed:
Add changelog and doc

pgrete committed:
Restore original logic

pgrete commented:
The failed tests were related to a change in logic for setting `dt` and `tlim` by default as those tests did not include a temporal driver.
I now reverted that logic to the original state (i.e., `dt=0` by default, which allows for the output being triggered).
I also added changelog and doc.
If all tests pass I'm now happy to merge.

BenWibking commented on doc/sphinx/src/outputs.rst:
```diff
@@ -6,10 +6,20 @@ Outputs
+
+- ``dt = 0.1`` means that the output for the block is written every 0.1
+  in simulation time.
+- ``dt = 100`` means that the output for the block is written every 100
```
I think this should be `dn = 100`.

Yurlungur replied:
I agree.

pgrete replied:
Done.

BenWibking reviewed (changes requested):
Typo in docs. Otherwise LGTM

Yurlungur reviewed (approved):
Other than the minor docs typo LGTM

pgrete committed:
Fix typo in doc
